tests/unit/test_charm_status.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import unittest
+from unittest.mock import patch
+
+import hypothesis.strategies as st
+from helpers import patch_network_get
+from hypothesis import assume, given
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+from charm import PrometheusCharm
+
+logger = logging.getLogger(__name__)
+
+
+class TestActiveStatus(unittest.TestCase):
+    """Feature: Charm's status should reflect the correctness of the config / relations.
+
+    Background: When launched on its own, the charm should always end up with active status.
+    In some cases (e.g. Ingress conflicts) the charm should go into blocked state.
+    """
+
+    def setUp(self) -> None:
+        self.app_name = "prometheus-k8s"
+
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.KubernetesServicePatch", lambda x, y: None)
+    @given(st.booleans(), st.integers(1, 5))
+    def test_unit_is_active_when_deployed_in_isolation(self, is_leader, num_units):
+        """Scenario: Unit is deployed without any user-provided config or regular relations.
+
+        The charm should end up in ActiveState when deployed in isolation with defaults only.
+        """
+        # without the try-finally, if any assertion fails, then hypothesis would reenter without
+        # the cleanup, carrying forward the units that were previously added
+        self.harness = Harness(PrometheusCharm)
+        self.peer_rel_id = self.harness.add_relation("prometheus-peers", self.app_name)
+
+        try:
+            self.assertEqual(self.harness.model.app.planned_units(), 1)
+
+            # GIVEN any number of units present
```

Comment:
It is not clear to me what is the significance of varying the number of units in this test. Prometheus only sets its IP address into the peer relation. I do not see how this affects its reload behaviour.